argusclient/client.py

```diff
@@ -438,13 +439,50 @@ def _init_all(self, coll=None):
                 self._coll[id] = perms
             self._retrieved_all = True
 
-    def get_permissions_for_entities(self, entityIds):
+    def get_permissions_for_entities(self, entity_ids):
         """
         Gets permissions that are associated with the given entity id's.
 
         :return: a dict of entity id's mapped to a list of :class:`argusclient.model.Permission` objects
         """
-        return convert(self.argus._request("post", "permission/entityIds", dataObj=entityIds))
+        entityIds = []
+        for entity_id in entity_ids:
+            if entity_id not in self._coll:
+                entityIds.append(entity_id)
```

Comment:
Is it filtering out entity id's for which the permissions have not yet been queried? To save processing time?

Reply:
yeah and not make the request again if we already have the info from a previous call for that entity ID made

Reply:
Ah ok